test/conformance/device_code/CMakeLists.txt

```diff
@@ -3,6 +3,37 @@
 # See LICENSE.TXT
 # SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
 
+set(AMD_ARCH "${UR_CONFORMANCE_AMD_ARCH}")
```

Comment:
I'm thinking when `AMD_ARCH` is already set in a `build/CMakeCache.txt` and `UR_CONFORMANCE_AMD_ARCH=""` we should use the value of `AMD_ARCH` but also emit a CMake warning saying its deprecated and to switch to `UR_CONFORMANCE_AMD_ARCH`.